### PR TITLE
chore: run Github Actions CI daily

### PR DESCRIPTION
**Pull request by nolanmar511**
I saw failures in https://github.com/google/pprof/pull/559 that didn't seem related to my change, and wanted to be able to check if these tests had passed recently. Running our CI daily will allow us to do this.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -4,6 +4,8 @@ on:
     branches:
       - master
   pull_request:
+  schedule:
+    - cron: '0 2 * * *' # Run everyday, at 2AM UTC.
```
Comment (aalexand) [inline .github/workflows/ci.yaml:8]: nit: "every day"?

**Review by aalexand — DISMISSED** [review]

**Review by wyk9787 — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (nolanmar511) [reply .github/workflows/ci.yaml]: Done.

**Comment by wyk9787** [comment]
Somehow two tests were failing and they seem transient. Restarted CI.

**Comment by nolanmar511** [comment]
I think I've seen the following failure a couple of times (in one of the macos tests):
```
==> Installing graphviz
==> Pouring graphviz-2.44.1.catalina.bottle.tar.gz
🍺  /usr/local/Cellar/graphviz/2.44.1: 506 files, 18MB
# runtime/cgo
ld: unsupported tapi file type '!tapi-tbd' in YAML file '/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libSystem.tbd' for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error: Process completed with exit code 2.
```

Hopefully they'll suceed when re-run; but I'm a little worried some of our tests may be broken. (One of my main motivations for sending this PR was to confirm test failures I saw on https://github.com/google/pprof/pull/559 were not related to my changes).

**Comment by aalexand** [comment]
Yes, I think running these tests regularly to deflake them was a good idea.

>

**Comment by nolanmar511** [comment]
I think the following tests are broken:
* ci / test-mac (1.13, macos-10.15, 10.3) (pull_request) 
* ci / test-mac (1.14, macos-10.15, 10.3) (pull_request) 

They fail with:
```
==> Installing graphviz
==> Pouring graphviz-2.44.1.catalina.bottle.tar.gz
🍺  /usr/local/Cellar/graphviz/2.44.1: 506 files, 18MB
# runtime/cgo
ld: unsupported tapi file type '!tapi-tbd' in YAML file '/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libpthread.tbd' for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error: Process completed with exit code 2.
```

Maybe we should stop install graphviz on tests using xcode 10.3. @wyk9787 -- What do you think?

**Comment by wyk9787** [comment]
> I think the following tests are broken:
> 
> * ci / test-mac (1.13, macos-10.15, 10.3) (pull_request)
> * ci / test-mac (1.14, macos-10.15, 10.3) (pull_request)
> 
> They fail with:
> 
> ```
> ==> Installing graphviz
> ==> Pouring graphviz-2.44.1.catalina.bottle.tar.gz
> 🍺  /usr/local/Cellar/graphviz/2.44.1: 506 files, 18MB
> # runtime/cgo
> ld: unsupported tapi file type '!tapi-tbd' in YAML file '/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libpthread.tbd' for architecture x86_64
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> Error: Process completed with exit code 2.
> ```
> 
> Maybe we should stop install graphviz on tests using xcode 10.3. @wyk9787 -- What do you think?

I wonder whether it makes sense to stop testing xcode 10.3 altogether. https://github.com/golang/go/wiki/MinimumRequirements hasn't been updated for quite some time.

**Comment by nolanmar511** [comment]
@wyk9787 -- Since pprof is vendored into golang, differing from https://github.com/golang/go/wiki/MinimumRequirements is really not adviseable. It means we will have breakages the next time pprof is vendored into golang, and those are not fun for us to deal with (we need to drop everything to get a fix) and not fun for those vendoring in pprof (they have to reach out to us and wait for a fix).

March 5th 2020 is a pretty recent update to that file, all things considered (thankfully for those of us using languages, the supported platforms don't change too often).

**Comment by aalexand** [comment]
> @wyk9787 -- Since pprof is vendored into golang, differing from https://github.com/golang/go/wiki/MinimumRequirements is really not adviseable. It means we will (likely) have breakages the next time pprof is vendored into golang, and those are not fun for us to deal with (we need to drop everything to get a fix) and not fun for those vendoring in pprof (they have to reach out to us and wait for a fix).
> 
> March 5th 2020 is a pretty recent update to that file, all things considered (thankfully for those of us using languages, the supported platforms don't change too often).
> 
> If we could skip install graphviz on that platform, tests which rely on graphviz won't be run. It's something we've done before.

The issue seems to be similar to https://trac.macports.org/ticket/60893, but there is no resolution there. I think stopping installing graphviz on this platform for now is reasonable.

**Comment by wyk9787** [comment]
> If we could skip install graphviz on that platform, tests which rely on graphviz won't be run. It's something we've done before.

SGTM. 

Will tests today skip themselves once they found graphviz is not installed or this is something we need to support additionally?

**Comment by nolanmar511** [comment]
> Will tests today skip themselves once they found graphviz is not installed or this is something we need to support additionally?

Yes; tests that depend on graphivz will already be skipped if graphviz is not installed.

**Comment by wyk9787** [comment]
> > Will tests today skip themselves once they found graphviz is not installed or this is something we need to support additionally?
> 
> Yes; tests that depend on graphivz will already be skipped if graphviz is not installed.

The issue seems to be incompatibility between macOS 10.15 and Xcode 10.3 since the issue persists when it tries to fetch other Go packages as well. 10.15 is the only macOS Github Action supports, so I think we should move tests for xcode 10.3 to Travis CI., which uses macOS 10.14 for xcode10.3: https://docs.travis-ci.com/user/reference/osx/.

WDYT?

**Comment by kalyanac** [comment]
Moving to Travis is probably the only option if we have to keep testing that Xcode version.

**Comment by kalyanac** [comment]
Its not clear to me why we need to test XCode 10.3. We are not testing on all the macOS versions listed at https://github.com/golang/go/wiki/MinimumRequirements

**Review by kalyanac — APPROVED** [review]
Skipping install on XCode 10.3 sounds good. We can avoid Travis CI test then.

**Review by wyk9787 — APPROVED** [review]

**Pull request merged**